packages/coil-extension/src/background/services/Stream.ts

```diff
@@ -327,68 +404,60 @@ export class Stream extends EventEmitter {
     })
   }
 
-  onMoney(sentAmount: string) {
+  async stop(): Promise<void> {
+    this._active = false
+    if (!this._connection) return
+
+    this._debug('initiating stream shutdown')
+    if (this._ilpStream.isOpen()) {
```

Comment:
I think at some point (not necessarily this pull request) we should have this factored so that there's no doubt that there's no interleaving of operations. 

Was hoping to pull out a shared set of classes can use in oauth-scripts and the extension.